paramiko_jump/util.py

```diff
@@ -0,0 +1,129 @@
+"""
+Utility functions and classes designed to help simplify paramiko jump as well as add extra features.
+
+"""
+from getpass import getpass
+
+from contextlib import contextmanager
+from typing import (
+    AnyStr,
+    Tuple,
+    Iterator,
+    Sequence,
+    Optional,
+    Callable,
+)
+from paramiko_jump.client import SSHJumpClient
+
+import paramiko
+
+
+Prompt = Tuple[AnyStr, bool]
+Optional[Callable] = None
+
+
+class StatefulAuthHandler:
+    """Stateful auth handler for paramiko that will return a list of auth
+     parameters for every CLI prompt
+
+    Example
+    -------
+        >>> from paramiko_jump.util import StatefulAuthHandler
+        >>> handler = StatefulAuthHandler(['password'],['1'])
+        >>> handler()
+        ['password']
+        >>> handler()
+        ['1']
+
+    Attributes
+    ----------
+    _iterator: Iterator
+        Iterator to iterate through all the objects in an iteratable object
+    """
+
+    _iterator = None
+
+    def __init__(self, *items: Sequence):
+        self._iterator = iter(items)
+
+    def __call__(self, *args: Sequence, **kwargs: Sequence):
```

Comment:
Same here on the annotations. I recommend dropping annotations on args/kwargs. They're idiomatic Python as-is without annotations.